### PR TITLE
feat: Configurable and more flexible binning for CKFPerformanceWriter

### PR DESCRIPTION
**Pull request by benjaminhuth**
This allows for a more flexible binning for the `PlotTools`, and exposes this to python. For example, a binning can be configured like this:

```python
binningCfg = {
    "Pt" : acts.examples.Binning("pT", list(np.logspace(-2,2,40))),
    "Eta" : acts.examples.Binning("eta", 40, -4, 4),
    "Phi" : acts.examples.Binning("phi", 100, -3.15, 3.15),
}

s.addWriter(
    acts.examples.CKFPerformanceWriter(
		...
        effPlotToolConfig = acts.examples.EffPlotToolConfig(self.binningCfg),
    )
)
```

### CODE REVIEW

File: Examples/Framework/include/ActsExamples/Utilities/Helpers.hpp [hunk 1]
```diff
@@ -29,16 +29,33 @@ namespace ActsExamples {
 
 namespace PlotHelpers {
 /// @brief Nested binning struct for booking plots
-struct Binning {
+class Binning {
+ public:
   Binning() = default;
```
Comment (benjaminhuth) [inline Examples/Framework/include/ActsExamples/Utilities/Helpers.hpp:34]:
```suggestion
  Binning() : m_bins({0.0}) {}
```

**Review by andiwand — DISMISSED** [review]
nice!

**Review by andiwand — APPROVED** [review]

**Review by andiwand — DISMISSED** [review]

**Comment by paulgessinger** [comment]
This has physmon failures.

File: Examples/Python/src/Output.cpp [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // This file is part of the Acts project.
 //
-// Copyright (C) 2021 CERN for the benefit of the Acts project
+// Copyright (C) 2023 CERN for the benefit of the Acts project
```
Comment (AJPfleger) [inline Examples/Python/src/Output.cpp:3]:
```suggestion
// Copyright (C) 2021-2023 CERN for the benefit of the Acts project
```

File: Examples/Framework/src/Validation/ResPlotTool.cpp [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // This file is part of the Acts project.
 //
-// Copyright (C) 2019 CERN for the benefit of the Acts project
+// Copyright (C) 2023 CERN for the benefit of the Acts project
```
Comment (AJPfleger) [inline Examples/Framework/src/Validation/ResPlotTool.cpp:3]:
```suggestion
// Copyright (C) 2019-2023 CERN for the benefit of the Acts project
```

File: Examples/Framework/src/Utilities/Helpers.cpp [hunk 4]
```diff
@@ -1,6 +1,6 @@
 // This file is part of the Acts project.
 //
-// Copyright (C) 2019 CERN for the benefit of the Acts project
+// Copyright (C) 2023 CERN for the benefit of the Acts project
```
Comment (AJPfleger) [inline Examples/Framework/src/Utilities/Helpers.cpp:3]:
```suggestion
// Copyright (C) 2019-2023 CERN for the benefit of the Acts project
```

File: Examples/Framework/include/ActsExamples/Utilities/Helpers.hpp [hunk 5]
```diff
@@ -1,6 +1,6 @@
 // This file is part of the Acts project.
 //
-// Copyright (C) 2019 CERN for the benefit of the Acts project
+// Copyright (C) 2023 CERN for the benefit of the Acts project
```
Comment (AJPfleger) [inline Examples/Framework/include/ActsExamples/Utilities/Helpers.hpp:3]:
```suggestion
// Copyright (C) 2019-2023 CERN for the benefit of the Acts project
```

**Review by AJPfleger — CHANGES_REQUESTED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078160) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078159) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078158) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078157) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078156) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078155) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078154) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078153) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078152) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078151) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078150) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078149) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078148) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078147) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078146) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078145) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32078144) |

**Pull request merged**